python/cuspatial/cuspatial/core/geoseries.py

```diff
@@ -768,63 +775,174 @@ def contains_properly(self, other, align=True):
-            # Not necessary for points.
-            result = cudf.DataFrame(
-                {"idx": point_indices, "pip": point_result}
-            )
-            # if the number of points in the polygon is equal to the number of
-            # points, then the requirements for `.contains_properly` are met
-            # for this geometry type.
-            df_result = (
-                result.groupby("idx").sum().sort_index()
-                == result.groupby("idx").count().sort_index()
+            return ContainsProperlyBinpred(self, other, align)()
+
+    def equals(self, other, align=True):
+        """Compute if a GeoSeries of features A is equal to a GeoSeries of
+        features B. Features are equal if their coordinates are equal.
+
+        Parameters
+        ----------
+        other
+            a cuspatial.GeoSeries
+        align=True
+            to align the indices before computing .contains or not. If the
+            indices are not aligned, they will be compared based on their
+            implicit row order.
+
+        Examples
+        --------
+        Test if two points are equal:
+        >>> point = cuspatial.GeoSeries(
+            [Point(0, 0)],
             )
-            point_result = cudf.Series(
-                df_result["pip"], index=cudf.RangeIndex(0, len(df_result))
+        >>> point2 = cuspatial.GeoSeries(
+            [Point(0, 0)],
             )
-            point_result.name = None
-        return point_result
+        >>> print(point.equals(point2))
+        0    True
+        dtype: bool
+
+        Test if two polygons are equal:
+        >>> polygon = cuspatial.GeoSeries(
+            [
+                Polygon([[0, 0], [1, 0], [1, 1], [0, 0]]),
+            ]
+        )
+        >>> polygon2 = cuspatial.GeoSeries(
+            [
+                Polygon([[0, 0], [1, 0], [1, 1], [0, 0]]),
+            ]
+        )
+        >>> print(polygon.equals(polygon2))
+        0    True
+        dtype: bool
+
+        Returns
+        -------
+
+        result : cudf.Series
+            A Series of boolean values indicating whether each feature in A
+            is equal to the corresponding feature in B.
+        """
+        return EqualsBinpred(self, other, align)()
+
+    def covers(self, other, align=True):
+        """Compute if a GeoSeries of features A covers a second GeoSeries of
+        features B. A covers B if no points on B lie in the exterior of A.
+
+        Parameters
+        ----------
+        other
+            a cuspatial.GeoSeries
+        align=True
+            align the GeoSeries indexes before calling the binpred
+
+        Returns
+        -------
+        result : cudf.Series
+            A Series of boolean values indicating whether each feature in the
+            input GeoSeries covers the corresponding feature in the other
+            GeoSeries.
+        """
+        return CoversBinpred(self, other, align)()
+
+    def intersects(self, other, align=True):
```

Comment:
After looking at all these predicates on GeoSeries, I realized we really should just start from a copy of the documentation from GeoPandas GeoSeries here.  e.g. for [intersects](https://geopandas.org/en/stable/docs/reference/api/geopandas.GeoSeries.intersects.html):

> Returns a Series of dtype('bool') with value True for each aligned geometry that intersects other.
>
> An object is said to intersect other if its boundary and interior intersects in any way with those of the other.
>
> The operation works on a 1-to-1 row-wise manner:
> ../../../_images/binary_op-01.svg

Reply:
I'm pretty much using GeoPandas as my source, but trying not to just copy everything outright. What do you suggest?

Reply:
Just make sure every operation has consistent docs in terms of what is documented, what is defined, and what terminology and mathematics are used in the definitions.